### DISCUSSION TITLE
Hit Test for Irregular Shapes

### DISCUSSION THREAD

**Discussion by maoyy** [AvaloniaUI/Avalonia#13570] [Ideas]
I am developing a graphics application using Avalonia, which has a mouse drag function. During the process of dragging graphics with the mouse, I need to obtain other graphics that have collided. Currently, I have found that Avalonia only has point hit testing. I hope to add the hit testing function for irregular graphics. Thank you very much

**Comment by timunie** [comment]
@maoyy our hit-testing is not there for this purpose. It exists in order to test if a pointer has hit the shape. 

Imo, while this is a valid request in your case, in 95% of each apps this is not even a thing. So I think this is something one should implement per App or in an own controls library rather than in Avalonia directly.

**Comment by timunie** [comment]
will mark this as an Idea to discuss about for now. If we decide its useful to add this API, this will be a feature request with proper description etc.

**Reply by dbriard** [reply]
On regards to hit test, having a hit test method that take rect instead of point would be useful to port code from WPF more easily.
